File: frontend/frontend-customer/src/app/models/Customer.ts

export class Customer{
  
    idNumber?: String
    name?: String
    lastname?: String
    email?: String
    creditCardNumber?: String
    
    constructor({idNumber, name, lastname, email, creditCardNumber}:{idNumber?: String ,name?: String, lastname?: String, email?: String, creditCardNumber?: String}) {
        this.idNumber = idNumber
        this.name = name
        this.lastname = lastname
        this.email = email
        this.creditCardNumber = creditCardNumber
    }

}

File: backend/Core/src/CoreEntities/src/Models/Customer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CoreEntities.src.Models
{
    [Table("tb_customer")]
    public class Customer
    {
        public int? Id { get; set; }
        public string? Name { get; set; }
        public string? LastName { get; set; }
        public string? Email { get; set; } 
        public string? IdNumber { get; set; }
        public string? CreditCardNumber { get; set; }

        public IEnumerable<Reservation>? Reservations { get; set; }
    }
}
